glusterd2/daemon/events.go

```diff
@@ -0,0 +1,33 @@
+package daemon
+
+import (
+	"strconv"
+
+	"github.com/gluster/glusterd2/glusterd2/events"
+)
+
+type daemonEvent string
+
+const (
+	daemonStarting       daemonEvent = "daemon.starting"
+	daemonStarted                    = "daemon.started"
+	daemonStartFailed                = "daemon.startfailed"
+	daemonStopping                   = "daemon.stopping"
+	daemonStopped                    = "daemon.stopped"
+	daemonStopFailed                 = "daemon.stopfailed"
+	daemonStartingAll                = "daemon.startingall"
+	daemonStartedAll                 = "daemon.startedall"
+	daemonStartAllFailed             = "daemon.startallfailed"
+)
+
+// newEvent returns an event of given type with daemon data filled
+func newEvent(d Daemon, e daemonEvent, pid int) *events.Event {
+	data := make(map[string]string)
```

Comment:
You can replace this sort of pattern with:

```
	data := map[string]string{
		"foo": "hey",
		"bar": "hey2",
	}
```

OR

just use a struct, which is probably more appropriate.

HTH

Reply:
Having a unique struct per event for additional data, would work out if the events were localized. It wouldn't work for global events, as the global recievers wouldn't know what struct was set to properly deserialize.

Reply:
I'll change to the suggested coding pattern though.